src/Runtime.Base/src/System/Runtime/CachedInterfaceDispatch.cs

```diff
@@ -27,7 +27,7 @@ private static IntPtr RhpCidResolve_Worker(object pObject, IntPtr pCell)
             DispatchCellInfo cellInfo;
 
             InternalCalls.RhpGetDispatchCellInfo(pCell, out cellInfo);
-            IntPtr pTargetCode = RhResolveDispatchWorker(pObject, (void*)pCell, ref cellInfo);
+            IntPtr pTargetCode = RhResolveDispatchWorker(new EETypePtr((IntPtr)pObject.EEType), (void*)pCell, ref cellInfo);
```

Comment:
This was passing the object around to make ICastable work. ICastable support was removed in https://github.com/dotnet/corert/pull/8103, but we have since then learned that we will need to add it back (in a slightly different form). It would be nice to keep passing object around to make that possible.

Can `RhResolveDispatchWorker` take both object and EETypePtr? 

Can the number of places that pass around the type be minimized somehow?